src/Topologies/equispacedgrids.jl

```diff
@@ -0,0 +1,93 @@
+"""
+    GridTopology1D(mesh)
+
+A line topology defined on an equispaced linear mesh.
+"""
+
+function GridTopology1D(mesh::Meshes.EquispacedLineMesh)
+    x1boundary = mesh.domain.x3boundary
+    x2boundary = nothing
+    boundaries =
+        isnothing(x1boundary) ? NamedTuple() : NamedTuple{x1boundary}((1, 2))
+    return TensorProductTopology(mesh, boundaries)
+end
+
+
+function vertex_coordinates(
+    topology::TensorProductTopology{M},
+    elem::Integer,
+) where {M <: EquispacedRectangleMesh}
+    @assert 1 <= elem <= nlocalelems(topology)
+
+    # convert to 0-based indices
+    mesh = topology.mesh
+    n1 = mesh.n1
+    n2 = mesh.n2
+    range1 = mesh.range1
+    range2 = mesh.range2
+
+    z2, z1 = fldmod(elem - 1, n1)
+
+    c1 = Geometry.Cartesian2DPoint(range1[z1 + 1], range2[z2 + 1])
+    c2 = Geometry.Cartesian2DPoint(range1[z1 + 2], range2[z2 + 1])
+    c3 = Geometry.Cartesian2DPoint(range1[z1 + 1], range2[z2 + 2])
+    c4 = Geometry.Cartesian2DPoint(range1[z1 + 2], range2[z2 + 2])
+    return (c1, c2, c3, c4)
+end
+
+function opposing_face(
```

Comment:
does this not work for all `TensorProductTopology`s?

Reply:
It would, but all the following algorithm becomes a one-liner in the `TensorProductTopology` case (i.e., `return topology.mesh.faces[(elem - 1) * 4 + face][3:5]`), because of how the data structure is set up in its constructor. So I thought it was better to take advantage of that. But if you think we want to unify this and do not dispatch on the concrete type, I can get rid of the special implementation, and just leave this redundant way for that data structure